From ce8c34e6efc671b1f8e19346cc9e199b972772db Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Marchesin?= <marcheu@chromium.org>
Date: Mon, 28 Jan 2013 14:05:42 -0800
Subject: [PATCH] glx: Check that swap_buffers_reply is non-NULL before using
 it

Check that the return value from xcb_dri2_swap_buffers_reply is
non-NULL before accessing the struct members.

Change-Id: Ia289c04ed1237326873e23c093675faa8708b557
---
 src/glx/dri2_glx.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/glx/dri2_glx.c b/src/glx/dri2_glx.c
index 12b3026..141af1c 100644
--- a/src/glx/dri2_glx.c
+++ b/src/glx/dri2_glx.c
@@ -787,9 +787,11 @@ dri2SwapBuffers(__GLXDRIdrawable *pdraw, int64_t target_msc, int64_t divisor,
        XSync(pdraw->psc->dpy, False);
        swap_buffers_reply =
           xcb_dri2_swap_buffers_reply(c, swap_buffers_cookie, NULL);
-       ret = merge_counter(swap_buffers_reply->swap_hi,
-                           swap_buffers_reply->swap_lo);
-       free(swap_buffers_reply);
+       if (swap_buffers_reply) {
+          ret = merge_counter(swap_buffers_reply->swap_hi,
+                              swap_buffers_reply->swap_lo);
+          free(swap_buffers_reply);
+       }
     }
 
     if (psc->show_fps) {
-- 
1.8.1

